### PR TITLE
Frontend for basic brush selector

### LINKED ISSUE #179: Allow users to label Scans with Brush
## Expected Behavior

User should be able to draw a Label with brush (in the same way as in Paint or any other graphics tool). User should control the size of the brush and modify his/her drawing with eraser.

### PR DESCRIPTION
**Pull request by Darkoon**
Fixes #179 

## Proposed Changes

  - Added specific model files hierarchy
  - Some more generic classes in legacy code
  - Brush selector without customization options

## UI changes (optional)

opera_2018-07-15_19-41-11

opera_2018-07-15_19-41-56

## Additional comment (optional)

To work with brush selections in most optimal way, selections are saved as one "snapshot" from canvas. As it is really handy way to deal with it that way, other selections must be cleared from canvas in process of painting selection and saving it. From business POV it seems OK, because when selecting something, we dont want any other selections to distract us. If it is not acceptable, I can rework it to use another canvas layer but it will add unnecessary complexity to brush selector. Lets discuss it.

### CODE REVIEW

File: frontend/src/environments/environment.local.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 export const environment = {
     production: false,
-    API_URL: 'http://localhost:51000/api/v1',
-    WEBSOCKET_URL: 'http://localhost:51001',
-    WEBSOCKET_PATH: '/socket.io'
+    API_URL: 'https://kask.eti.pg.gda.pl/medtagger/api/v1',
```
Comment (jszemplinski) [inline frontend/src/environments/environment.local.ts]: ;)

**Comment by jpowie01** [comment]
Works pretty well! Changing browser's window and scaling works well even on Safari! <3 I think it's fine for now to disappear other Labels from canvas - that works pretty fine and is somehow intuitive. We can improve it later ;)

But... I think that there should be only one brush element per Slice for given Tag. This way user will be able to edit the entered Label and do not create new ones by mistake! What's more, eraser should be easier to implement with this design. What do you think? We can discuss it on Discord if you would like to ;)

**Example:** User would like to label ribs with brush on a single Slice. This is the final effect (5 Labels instead of one):
screen shot 2018-07-17 at 23 50 02

There is no point to differ between ribs because the main information about ribs is decoded per pixel on image. It doesn't matter if it's a single image or five of them from the final Label perspective. Apart from that we will have 5x less data on the backend side! :)

**Comment by Darkoon** [comment]
@jpowie01 Great idea :) How about flexibility here? Do we want to stick to the idea of single selection per slice? Or should it be an option for selecting (multi selection/single selection per slice)?

**Comment by jpowie01** [comment]
In case of Brush selector, I would stick to a single selection per Slice (**for given Tag!**). But we can discuss it today on Scrum :)

**Comment by jpowie01** [comment]
I've run it on my local dev instance and it works pretty cool! ...but I found some Issues 😔

Here are my reproductions:

----------------------------------------

### First Issue
1. Use Scan Category with two Label Tags (eg. Left Kidney, Right Kidney).
2. Select first Label Tag (Left Kidney) and select Brush selector.
3. Draw something on the Scan.
4. Switch to the second Label Tag (Right Kidney) and select Brush selector.
5. Draw something on the Scan.
6. Click on the 'eye' button for the Label Selection of Left Kidney.
7. Boom! Nothing happens...
8. Click on the 'eye' button for the Label Selection of Right Kidney.
9. Boom! Both of your drawings disappeared...

screen shot 2018-08-01 at 21 42 03

It seems that both of these drawings were made on the same canvas which is bad... There should be a separate canvas for each **Label Tag**.

----------------------------------------

### Second Issue
Another issue that I found:

1. Use Scan Category with at least one Label Tag (eg. Nodule).
2. Select Label Tag (Nodule) and select Brush selector.
3. Draw something on the Scan (and release your mouse!).
4. Draw something on the Scan (and release your mouse!).
5. Draw something on the Scan (and release your mouse!).
6. Draw something on the Scan (and release your mouse!).
7. Draw something on the Scan (and release your mouse!).
8. Boom! ID of the Label Selection has increased to 5.

It may be confusing for users... :(

----------------------------------------

### Third Issue

This is minor and may be fixed later:

1. Use Scan Category with at least one Label Tag (eg. Nodule).
2. Select Label Tag (Nodule) and select Brush selector.
3. Draw something on the Scan.
4. Move to the next Slice.
5. Click on the 'pin' button for the Label Selection of Nodule.
6. Boom! Selection is red :)

Do the same for other Selectors and all of them will be blue :)

File: frontend/src/app/pages/marker-page/marker-page.component.html [hunk 2]
```diff
@@ -15,7 +15,7 @@ <h3>Available Tags</h3>
             <h3>Available Tools</h3>
             <div class="buttons">
                 <button mat-mini-fab [disabled]="!isToolSupportedByCurrentTag('RECTANGLE')" (click)="setSelector('RECTANGLE')"><mat-icon [svgIcon]="getToolIconName('RECTANGLE')"></mat-icon></button>
-                <button mat-mini-fab disabled><mat-icon [svgIcon]="getToolIconName('BRUSH')"></mat-icon></button>
+                <button mat-mini-fab [disabled]="!isToolSupportedByCurrentTag('BRUSH')"><mat-icon [svgIcon]="getToolIconName('BRUSH')"></mat-icon></button>
```
Comment (jpowie01) [inline frontend/src/app/pages/marker-page/marker-page.component.html]: Remember to add `(click)="setSelector('BRUSH')"` :)

File: frontend/src/environments/environment.local.ts (hunk 1 above)
Comment (kolszewska) [reply frontend/src/environments/environment.local.ts]: @Darkoon ? :)

Comment (Darkoon) [reply frontend/src/environments/environment.local.ts]: @kolszewska old but gold

**Pull request merged**